build/js/live-editor.output_pjs.js

```diff
@@ -1152,6 +1152,127 @@ var BabyHint = {
 // TODO(jlfwong): Stop globalizing BabyHint
 window.BabyHint = BabyHint;
 
+var PJSResourceCache = Backbone.Model.extend({
+
+    initialize: function(options) {
+        this.canvas = options.canvas;   // customized Processing instance
+        this.output = options.output;   // LiveEditorOutput instance
+        this.cache = {};
+        this.imageHolder = null;
+    },
+
+    // Load and cache all resources (images and sounds) that could be used in 
+    // the environment.  Right now all resources are loaded as we don't have 
+    // more details on exactly which images will be required.
+    //
+    // Execution is delayed once a getImage/getSound appears in the source code
+    // and none of the resources are cached. Execution begins once all the
+    // resources have loaded.
+    cacheResources: function(userCode, callback) {
+        var getImageRegex = /getImage\s*\(['"](.*?)['"]\)/g;
+        var images = this.getRegexMatches(userCode, getImageRegex, 1);
+        var getSoundRegex = /getSound\s*\(['"](.*?)['"]\)/g;
+        var sounds = this.getRegexMatches(userCode, getSoundRegex, 1);
+
+        // Insert the images into a hidden div to cause them to load
+        // but not be visible to the user
+        if (!this.imageHolder) {
+            this.imageHolder = $("<div>")
+                .css({
+                    height: 0,
+                    width: 0,
+                    overflow: "hidden",
+                    position: "absolute"
+                })
+                .appendTo("body");
+        }
+
+        var promises = images.map(this.loadImage.bind(this));
+        promises = promises.concat(sounds.map(this.loadSound.bind(this)));
```

Comment:
I was thinking of having a single array of all resources to load with each item being an object with the filename and filetype and then this would turn into a single array mapped against ```loadResource``` which would multiplex between ```loadImage``` and ```loadSound```.  I'm going to give that a try to see that's any better.